src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/InitializeHandler.cs

```diff
@@ -16,7 +16,12 @@ internal class InitializeHandler : IRequestHandler<InitializeParams, InitializeR
         {
             Capabilities = new ServerCapabilities
             {
-                // Placeholder until more capabilities are enabled
+                CompletionProvider = new CompletionOptions()
+                {
+                    AllCommitCharacters = new[] { " ", ".", ";", ">", "=" },
```

Comment:
>AllCommitCharacters = new[] { " ", ".", ";", ">", "=" }, [](start = 20, length = 56)

Is this going to change to eventually call into the html/C# language servers?

Reply:
Don't plan to. We want to control the specific experience in Razor so these represent how HTML should work in Razor scenarios even if that slightly diverges from HTML outside of  Razor. If HTML ends up adding more and we want to consume that new functionality we'd update this list.

The motivation behind this decision is "less is more". Trying to avoid the same projection issues we've had in the past where we're too allowing of all features that can happen in a Razor document and those features don't always make sense for Razor. In the case that they do, we'll white list them.

Reply:
We should chat offline about this. If I want to add CSS completion this is going to affect this list. If we add a new CSS completion provider, that might change this list. I think you should own turning on/off features, but not decide how those features work.

---
In reply to: [392324289](https://github.com/dotnet/aspnetcore-tooling/pull/1685#discussion_r392324289) [](ancestors = 392324289)

Reply:
@NTaylorMullen should we have two separate TriggerCharacter list (One for HTML and one for C#)? I ask because we can just no-op in cases when our completion handler gets called with a C# trigger character and our projection result says we are in HTML context. Do we do this in VSCode?

Reply:
We don't do it in VSCode but it might make sense to start to prevent having unnecessary asks for completions. To do it we'd need to have a total of three language servers instead of the two today:
1. rzls
2. HtmlMiddleman
3. CSharpMiddleman